electron/src/renderer/menu/context.ts

```diff
@@ -74,63 +114,38 @@ const imageMenu: ElectronMenuWithImageAndTime = Menu.buildFromTemplate([
   },
 ]);
 
-window.addEventListener(
-  'contextmenu',
-  event => {
-    const element = event.target as HTMLElement;
-
-    copyContext = '';
-
-    if (element.nodeName === 'TEXTAREA' || element.nodeName === 'INPUT') {
-      event.preventDefault();
-
-      createTextMenu();
-      textMenu.popup({window: remote.getCurrentWindow()});
-    } else if (element.classList.contains('image-element') || element.classList.contains('detail-view-image')) {
-      event.preventDefault();
-      const elementSource = (element as HTMLImageElement).src;
-      const parentElement = element.closest('.message-body') as HTMLDivElement;
-      const timeElement = parentElement.getElementsByTagName('time')[0];
-      if (timeElement) {
-        const imageTimestamp = timeElement.dataset.timestamp;
-        imageMenu.timestamp = imageTimestamp;
-      }
-      imageMenu.image = elementSource;
-      imageMenu.popup({window: remote.getCurrentWindow()});
-    } else if (element.nodeName === 'A') {
-      event.preventDefault();
-
-      const elementHref = (element as HTMLLinkElement).href;
-      copyContext = elementHref.replace(/^mailto:/, '');
-      defaultMenu.popup({window: remote.getCurrentWindow()});
-    } else if (element.classList.contains('text')) {
-      event.preventDefault();
-
-      copyContext = (window.getSelection() || '').toString() || element.innerText.trim();
-      defaultMenu.popup({window: remote.getCurrentWindow()});
-    } else {
-      // Maybe we are in a code block _inside_ an element with the 'text' class?
-      // Code block can consist of many tags: CODE, PRE, SPAN, etc.
-      let parentNode = element.parentNode;
-      while (parentNode && parentNode !== document && !(parentNode as HTMLElement).classList.contains('text')) {
-        parentNode = parentNode.parentNode;
-      }
-      if (parentNode !== document) {
-        event.preventDefault();
-        copyContext = (window.getSelection() || '').toString() || (parentNode as HTMLElement).innerText.trim();
-        defaultMenu.popup({window: remote.getCurrentWindow()});
-      }
+remote.getCurrentWebContents().on('context-menu', (_event, params) => {
+  const window = remote.getCurrentWindow();
+
+  if (params.isEditable) {
+    const textMenu = createTextMenu(params);
+    textMenu.popup({window});
+  } else if (params.mediaType === 'image') {
+    imageMenu.image = params.srcURL;
+    imageMenu.popup({window});
+  } else if (!!params.linkURL) {
+    const copyContext = params.linkURL.replace(/^mailto:/, '');
+    createDefaultMenu(copyContext).popup({window});
+  } else if (!!params.selectionText) {
+    const copyContext = params.selectionText;
+    createDefaultMenu(copyContext).popup({window});
+  } else if (params.editFlags.canCopy) {
```

Comment:
This condition does the same as `!!params.selectionText`. Can we put them in one if block connected with a or condition?